### PR TITLE
Add multi-tenant mode to ASOv2 helm chart

### PR DESCRIPTION
**Pull request by super-harsh**
Closes: https://github.com/Azure/azure-service-operator/issues/2118

**What this PR does / why we need it:**

This PR adds script and changes to support multitenant mode in helm chart for ASOv2

**Special notes for your reviewer:**

Currently, have added the generated files for review. Will be removing them before merge.
Will be adding usage docs as well before merging.

### CODE REVIEW

File: v2/charts/azure-service-operator/values.yaml [hunk 1]
```diff
@@ -73,3 +73,14 @@ installCRDs: true
...
 podAnnotations: {}
+
+# multiTenant contains the value to enable multi-tenant mode for ASOv2. Also, it contains 'isCluster' flag which is
+# used to know whether to install cluster or a tenant. If multiTenant.enable is true and isCluster set to false, chart will
+# install a tenant.
+multiTenant:
+  enable: false
+#  NOTE: If we set 'isCluster' to false, tenant mode will be enabled and have to make sure we have above 'installCRDs' also set to false
+#  while tenant installation.
+  isCluster: false
```
Comment (matthchr) [inline v2/charts/azure-service-operator/values.yaml:84]: I didn't say this in the other PR, but I wonder if we can come up with a better name for this option... something like `clusterScopeOnly` or `webhookOnly`?

Actually now that I think about it, we already have an `azureOperatorMode`, how does this option play with that? Is it possible that we should just use that option to manage the multitenancy?

**Review by matthchr — COMMENTED** [review]
See also my comments on the other Helm PR. I think I like this approach more, but curious what others think

File: v2/charts/azure-service-operator/values.yaml (hunk 1 above)
Comment (super-harsh) [reply v2/charts/azure-service-operator/values.yaml:84]: We van just use `azureOperatorMode` option to manage multi tenancy. However, it'll not be very much clear to the end user. Where this approach just uses bools and does the job below with `azureOperatorMode`


https://github.com/Azure/azure-service-operator/pull/2460/files#diff-197e51ba2031652d2fbef03840580aaa4b78b41915813efc25a14137e53c6509R9

**Comment by theunrepentantgeek** [comment]
I'm in favour of having a single template that gets compiled into multiple outputs. 

This mitigates against issues where we need to make an update and it only gets made in one of multiple templates, resulting in inconsistencies, ambiguities, and bugs.

File: v2/charts/azure-service-operator/values.yaml [hunk 2]
```diff
@@ -73,3 +73,12 @@ installCRDs: true
 
 # podAnnotations contain the pod annotations for Azure Service Operator
 podAnnotations: {}
+
+# multiTenant contains the value to enable multi-tenant mode for ASOv2. Also, it contains 'isCluster' flag which is
+# used to know whether to install cluster or a tenant. If multiTenant.enable is true and isCluster set to false, chart will
+# install a tenant.
+multiTenant:
```
[the remark this answers is not in the document]
Comment (matthchr) [reply v2/charts/azure-service-operator/values.yaml:80]: I don't think this was fixed?

**Review by matthchr — APPROVED** [review]
LGTM, with 1 minor casing comment

File: v2/charts/azure-service-operator/values.yaml (hunk 2 above)
[the remark this answers is not in the document]
Comment (super-harsh) [reply v2/charts/azure-service-operator/values.yaml:80]: OOPS! My bad. Pushing a fix.

**Review by matthchr — APPROVED** [review]

**Pull request merged**